lldb/source/Host/common/Editline.cpp

```diff
@@ -700,12 +707,14 @@ unsigned char Editline::EndOrAddLineCommand(int ch) {
     }
   }
   MoveCursor(CursorLocation::EditingCursor, CursorLocation::BlockEnd);
-  fprintf(m_output_file, "\n");
+  LockedStreamFile locked_stream = m_output_stream_sp->Lock();
```

Comment:
And I suspect most/all of these callbacks are called from `GetLine` which means that they could reuse the `locked_stream` member instead of locking it once again.